lib/Sema/TypeCheckType.cpp

```diff
@@ -3022,6 +3022,14 @@ Type TypeChecker::substMemberTypeWithBase(ModuleDecl *module,
   }
 
   if (auto *aliasDecl = dyn_cast<TypeAliasDecl>(member)) {
+    // FIXME: If this is a protocol typealias and we haven't built the
+    // protocol's generic environment yet, do so now, to ensure the
+    // typealias's underlying type has fully resoved dependent
+    // member types.
+    if (auto *protoDecl = dyn_cast<ProtocolDecl>(aliasDecl->getDeclContext()))
```

Comment:
Should this `dyn_cast` be `getAsProtocolOrProtocolExtensionContext`? (i.e. do we need to handle extensions?)